libbeat/beat/beat.go

```diff
@@ -218,7 +217,7 @@ func (b *Beat) Run() error {
 	// Callback is called if the processes is asked to stop.
 	// This needs to be called before the main loop is started so that
 	// it can register the signals that stop or query (on Windows) the loop.
-	service.HandleSignals(b.Exit)
+	service.HandleSignals(b.Stop)
```

Comment:
Exit was put here on purpose as Stop could also be used to later use again Start. In the case of exit, the complete beat exits which I think should be the case when CTRL-C is used. If Exit is used, the b.exit channel is probably never closed. Let me see if we can find an alternative solution here.